### PR TITLE
Use Xmacro to centralize selectors definition

### PR DESCRIPTION
**Pull request by fbeutin-ledger**
Pros:
- 3 less `EDIT THIS`
- No risk of mismatch between enum order and array order
- No risk of badly defining the NUMBER

Con:
- might be a bit less explicit ?

### CODE REVIEW

File: src/boilerplate_plugin.h [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
-// EDIT THIS: Change the naming (`selector_t`), and add your selector names.
-typedef enum {
-    SWAP_EXACT_ETH_FOR_TOKENS = 0,
-    BOILERPLATE_DUMMY_2,
+// All possible selectors of your plugin.
+// EDIT THIS: Enter your selectors here, in the format X(NAME, value)
+// A macro below will create for you:
+//     - an enum named selector_t with every NAME
+//     - a map named SELECTORS associating each NAME with it's value
+#define XMACRO(X)                            \
```
Comment (tdejoigny-ledger) [inline src/boilerplate_plugin.h]: name of this macro shall be more explicit, something like `SELECTOR_DEFINITION` for example

File: src/xmacro_helpers.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
```
Comment (tdejoigny-ledger) [inline src/xmacro_helpers.h:1]: Do we really want to create an helper for macros which will be probably used only for selectors definition ?

File: src/boilerplate_plugin.h (hunk 1 above)
Comment (fbeutin-ledger) [reply src/boilerplate_plugin.h]: used `SELECTORS_LIST`

File: src/xmacro_helpers.h (hunk 2 above)
Comment (fbeutin-ledger) [reply src/xmacro_helpers.h:1]: IMO it's utility noise that should not clutter the useful info stored in `plugins.h`

**Comment by fbeutin-ledger** [comment]
https://github.com/LedgerHQ/app-plugin-boilerplate/pull/103

**Pull request closed without merging**